### PR TITLE
Fixes Several Holodeck Infinite Materials Exploits

### PR DESCRIPTION
**Pull request by Kylerace**
## About The Pull Request
Fixes infinite material exploits for the following objects spawned into the holodeck: books, morgues, crayon boxes, boxes, and washing machines

## Why It's Good For The Game
exploits bad mkay

## Changelog
:cl:
fix: its now slightly harder to abuse the holodeck to break the second law of thermodynamics
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
aww, I liked being able to farm the holodeck for cardboard as a cardboard golem

**Comment by Kylerace** [comment]
> aww, I liked being able to farm the holodeck for cardboard as a cardboard golem

that seems incredibly unfortunate for you then

File: code/game/machinery/washing_machine.dm [hunk 1]
```diff
@@ -349,7 +349,8 @@ GLOBAL_LIST_INIT(dye_registry, list(
 		update_icon()
 
 /obj/machinery/washing_machine/deconstruct(disassembled = TRUE)
-	new /obj/item/stack/sheet/metal(drop_location(), 2)
+	if (!(flags_1 & HOLOGRAM_1))
+		new /obj/item/stack/sheet/metal(drop_location(), 2)
```
Comment (coiax) [inline code/game/machinery/washing_machine.dm:353]: Doesn't the holodeck already apply the NODECONSTRUCT flag that we could hook in here instead?

Comment (Kylerace) [reply code/game/machinery/washing_machine.dm]: That seems like a better solution for machinery/structures, done

**Pull request merged**